src/utils/suggestions.js

```diff
@@ -109,22 +120,29 @@ const getSuggestions = (
 					const val = extractSuggestion(label);
 					if (val) {
 						if (Array.isArray(val)) {
-							val.forEach(suggestion =>
-								populateSuggestionsList(suggestion, parsedSource, source));
+							val.forEach(suggestion => populateSuggestionsList(suggestion, parsedSource, source));
 						} else {
 							populateSuggestionsList(val, parsedSource, source);
+							return true;
 						}
 					}
 				}
 			}
 		}
+		return false;
 	};
 
-	suggestions.forEach((item) => {
-		fields.forEach((field) => {
-			parseField(item, field);
+	if (showDistinctSuggestions) {
+		suggestions.forEach((item) => {
+			fields.every(field => parseField(item, field));
```

Comment:
@jyash97 do you think we need to use `.some` here. because we need a distinct suggestion, hence if at least one is found we return that?

Reply:
Thanks for catching it. I have fixed now.